library/std/src/os/unix/fs.rs

```diff
@@ -1100,3 +1100,36 @@ pub fn lchown<P: AsRef<Path>>(dir: P, uid: Option<u32>, gid: Option<u32>) -> io:
 pub fn chroot<P: AsRef<Path>>(dir: P) -> io::Result<()> {
     sys::fs::chroot(dir.as_ref())
 }
+
+/// Create fifo at the specified path with the specified mode.
+///
+/// # Examples
+///
+/// ```
+/// # #![feature(unix_mkfifo)]
+/// # use std::{
+/// #     os::unix::fs::{mkfifo, PermissionsExt},
+/// #     fs::{File, Permissions, remove_file},
+/// #     io::{Write, Read},
+/// # };
+/// # fn main() -> std::io::Result<()> {
+/// # remove_file("/tmp/fifo")?;
+/// mkfifo("/tmp/fifo", Permissions::from_mode(0o774))?;
```

Comment:
If this test is to be run (I think it's a good idea to have some form of test), it may be better to call `mkdtemp` and create the FIFO there to ensure it doesn't interfere with anyone who actually has a `/tmp/fifo`.

Reply:
I'm not sure if we can use `libc::mktemp` here...honestly maybe we should also add a function for that

Reply:
Agreed, it would be nice to have. `libc` isn't available but the binding is pretty easy, the setup is a bit clunky but I think it's worth avoiding any possible issues

```rust
/// # // Use a temporary file to avoid conflicts or cleanup issues
/// # extern "C" { fn mktemp(template: *mut c_char) -> *mut c_char; }
/// # let mut buf = b"rust-fifo-XXXXXX\0";
/// # let fname = unsafe {
/// #     let ptr = mktemp(buf.as_ptr().cast::<c_char>()) };
/// #     assert!(!ptr.is_null(), "could not create tempfile");
/// #     CStr::from_ptr(ptr).to_str().unwrap()
/// # };
/// mkfifo(fname, Permissions::from_mode(0o774))?;
/// // ...
```

The rest LGTM

Reply:
My problem with this is portability, not sure if it would work on musl, and probably android and other libc reliably.

If we want to do this, might be easier to random generate the filename using [`std::random`](https://doc.rust-lang.org/std/random/index.html)?

Reply:
Fair point. We have tempfile for the std testsuite, maybe just make the doc example no_run and then add the actual test there?

Reply:
Yeah that makes more sense, I would do that

Reply:
I see this was made no_run but it doesn't look like a test wasn't added. If you don't have it yet, just copy+paste this example to std/src/os/unix/fs/tests.rs using a file name from `crate_helpers::tmpdir().join("fifo")`.

Reply:
Oops, I put it in `std/tests/mkfifo.rs` and forgot to `git add` it

Reply:
@tgross35 I have updated `std/src/os/unix/fs/tests.rs`